internal/auth/query.go

```diff
@@ -0,0 +1,844 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package auth
+
+const (
+	estimateCountAuthMethodsQuery = `
+select sum(reltuples::bigint) as estimate from pg_class where oid in (
+    'auth_password_method'::regclass,
+    'auth_ldap_method'::regclass,
+    'auth_oidc_method'::regclass
+)
+`
+
+	listDeletedIdsQuery = `
+select public_id
+  from auth_password_method_deleted
+ where delete_time >= @since
+ union
+select public_id
+  from auth_oidc_method_deleted
+ where delete_time >= @since
+ union
+select public_id
+  from auth_ldap_method_deleted
+ where delete_time >= @since
+`
+
+	listAuthMethodsTemplate = `
+with auth_methods as (
+    select public_id
+      from auth_method
+     where %s -- search condition for scope IDs is constructed
+  order by create_time desc, public_id asc
+     limit %d
+),
+ldap as (
+    select *
+      from ldap_auth_method_with_value_obj
+     where public_id in (select public_id from auth_methods)
+),
+oidc as (
+    select *
+      from oidc_auth_method_with_value_obj
+     where public_id in (select public_id from auth_methods)
+),
+password as (
+    select *
+      from auth_password_method_with_is_primary
+     where public_id in (select public_id from auth_methods)
+),
+final as (
+    select public_id,
+           scope_id,
+           case when ldap.is_primary_auth_method is true then
+             'ldap'
+           else null end
+           as primary_auth_method_subtype,
```

Comment:
Can't we just leave this as a boolean? This seems unnecessarily complicated
```suggestion
           is_primary_auth_method,
```
If the ldap registered method is called, it'll be because the subtype is ldap, and this will be true or false to indicate whether it's the primary auth method, no?

Reply:
There is the possibility of an auth method being stored for a certain subtype while still not being stored as the primary auth method, such as in the sqltests:

`
boundary=# select public_id, scope_id, name from auth_method;
  public_id   |   scope_id   |           name           
--------------+--------------+--------------------------
 apm___colors | o_____colors | Colors Auth Password
 apm_foodtruc | o__foodtruck | Food Truck Auth Password
(2 rows)

boundary=# select is_primary_auth_method, public_id, scope_id, name from auth_password_method_with_is_primary;
 is_primary_auth_method |  public_id   |   scope_id   |           name           
------------------------+--------------+--------------+--------------------------
 f                      | apm___colors | o_____colors | Colors Auth Password
 f                      | apm_foodtruc | o__foodtruck | Food Truck Auth Password
(2 rows)
`

Reply:
I'm not sure why that matters, it would just show up as false here, no?

Reply:
Got it, misunderstood the purpose of that column